pkg/cmd/start/provision.go

```diff
@@ -0,0 +1,80 @@
+package start
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/global"
+	"google.golang.org/grpc/codes"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+func provisionOwnJaeger(ctx context.Context, cl client.Client, ns string) {
+	tracer := global.TraceProvider().GetTracer(v1.BootstrapTracer)
+	ctx, span := tracer.Start(ctx, "provisionOwnJaeger")
+	defer span.End()
+
+	// this will provision a simplest instance, with in-memory storage
+	// for any other usage, we recommend users to create their own CRs
+	j := provisionedCR(types.NamespacedName{Name: "jaeger", Namespace: ns})
```

Comment:
should we name it `jaeger-operator` to make clear it is for use with the operator? Also `jaeger` would be a common name given to a default jaeger instance used by the applications (albeit in a different namespace).

Reply:
Not `jaeger-operator`, as there will already be a deployment named `jaeger-operator` in that namespace ;-)

Agree that the name is too generic, but on a second thought, perhaps we want to recommend users to disable the auto-provisioning by default? I think it's useful for development/ephemeral installations, but for production, I think people might want to use a properly provisioned Jaeger installation. In addition to that, not sure what people will think when they see something other than operators inside their operators namespace (`openshift-operators`, `marketplace`, ...).

Reply:
Agree - in which case is auto-provisioning really necessary? i.e. maybe better that user is responsible for provisioning the jaeger instance, so we don't need to worry about disabling this part. Even if tracing is enabled, it will just get reported no where.

But how will a user configure the jaeger instance to report to? i.e. the agent's collector URL?

Reply:
> Agree - in which case is auto-provisioning really necessary?

Good point, I'll remove this feature.

> the agent's collector URL?

Yes.

Reply:
Why I mean, is that currently they would have to change the collector URL to point to their existing jaeger instance - but when installed via OLM, this won't be possible?

Reply:
Until operators are configurable via OLM, we have to ship an operator configuration that can't be changed, possibly providing instructions on how to add a custom operator source with the alternate configuration. For non-OLM deployments, the `operator.yaml` can be changed directly.